components/property_access.rst

```diff
@@ -365,8 +371,228 @@ You can also mix objects and arrays::
+method should be called when a get, set, add or remove operation is needed on the
+property.
+
+.. configuration-block::
+
+    .. code-block:: php-annotations
+
+        // ...
+        use Symfony\Component\PropertyAccess\Annotation\Property;
+
+        class Person
+        {
+            /**
+             * @Property(getter="getFullName", setter="setFullName")
+             */
+            private $name;
+
+            /**
+             * @Property(adder="addNewChild", remover="discardChild")
+             */
+            private $children;
+
+            public function getFullName()
+            {
+                return $this->name;
+            }
+
+            public function setFullName($fullName)
+            {
+                $this->name = $fullName;
+            }
+        }
+
+    .. code-block:: yaml
+
+        # src/AppBundle/Resources/config/property_access.yml
+        Person:
+            name:
+                getter: getFullName
+                setter: setFullName
+            children:
+                adder:   addNewChild
+                remover: discardChild
+
+    .. code-block:: xml
+
+        <!-- src/AppBundle/Resources/config/property_access.xml -->
+        <?xml version="1.0" ?>
+
+        <property-access xmlns="http://symfony.com/schema/dic/property-access-mapping"
+                            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                            xsi:schemaLocation="http://symfony.com/schema/dic/property-access-mapping http://symfony.com/schema/dic/property-access-mapping/property-access-mapping-1.0.xsd">
+
+            <class name="Person">
+                <property name="name" getter="getFullName" setter="setFullName" />
+                <property name="children" adder="addNewChild" remover="discardChild" />
+            </class>
+
+        </property-access>
+
+Then, using the overriden methods is automatic:
+
+.. code-block:: php
+
+    $person = new Person();
+
+    $accessor->setValue($person, 'name', 'John Doe');
+    // will call setFullName
+
+    var_dump('Hello '.$accesor->getValue($person, 'name'));
+    // will return 'Hello John Doe'
+
+You can also associate a particular method with an operation on a property
+using the ``@PropertyGetter``, ``@PropertySetter``, ``@PropertyAdder`` and
+``@PropertyRemover`` annotations. All of them take only one parameter: ``property``.
+
+This allows creating virtual properties that are not directly stored in the
+object:
+
+.. configuration-block::
+
+    .. code-block:: php-annotations
+
+        // ...
+        use Symfony\Component\PropertyAccess\Annotation\PropertyGetter;
+        use Symfony\Component\PropertyAccess\Annotation\PropertySetter;
+
+        class Invoice
+        {
+            private $quantity;
+
+            private $pricePerUnit;
+
+            // Notice that there is no real "total" property
+
+            /**
+             * @PropertyGetter(property="total")
```

Comment:
@HeahDude, we could rename this annotation to just `Getter` and do the same to the others (`Setter`, `Adder` and `Remover`). Do you agree?